### ISSUE TITLE
3.4.0 TyperError of undefined path in DefaultLocalizer

### ISSUE THREAD

**Issue by bobzsj87** [Microsoft/BotBuilder#1540]
Due to the lowercase problem I downgraded to 3.2.3 and it worked fine. As it is fixed in 3.4.0, I upgrade directly without any other changes but it throws error and Messenger feeds back directly "default_error"

```
TypeError: Path must be a string. Received undefined
    at assertPath (path.js:7:11)
    at Object.join (path.js:466:7)
    at DefaultLocalizer.loadLocalePath (C:\Users\Bob PC\Dev\hunt\node_modules\botbuilder\lib\DefaultLocalizer.js:119:24)
    at C:\Users\Bob PC\Dev\hunt\node_modules\botbuilder\lib\DefaultLocalizer.js:104:27
    at C:\Users\Bob PC\Dev\hunt\node_modules\async\lib\async.js:181:20
    at iterate (C:\Users\Bob PC\Dev\hunt\node_modules\async\lib\async.js:262:13)
    at Object.async.forEachOfSeries.async.eachOfSeries (C:\Users\Bob PC\Dev\hunt\node_modules\async\lib\async.js:281:9)
    at Object.async.forEachSeries.async.eachSeries (C:\Users\Bob PC\Dev\hunt\node_modules\async\lib\async.js:214:22)
    at C:\Users\Bob PC\Dev\hunt\node_modules\botbuilder\lib\DefaultLocalizer.js:103:23
    at tryCallTwo (C:\Users\Bob PC\Dev\hunt\node_modules\promise\lib\core.js:45:5)
    at doResolve (C:\Users\Bob PC\Dev\hunt\node_modules\promise\lib\core.js:200:13)
    at new Promise (C:\Users\Bob PC\Dev\hunt\node_modules\promise\lib\core.js:66:3)
    at DefaultLocalizer.loadLocale (C:\Users\Bob PC\Dev\hunt\node_modules\botbuilder\lib\DefaultLocalizer.js:102:28)
    at C:\Users\Bob PC\Dev\hunt\node_modules\botbuilder\lib\DefaultLocalizer.js:57:19
    at C:\Users\Bob PC\Dev\hunt\node_modules\async\lib\async.js:181:20
    at Object.async.forEachOf.async.eachOf (C:\Users\Bob PC\Dev\hunt\node_modules\async\lib\async.js:233:13)
```

**Comment by aravindkarnam** [comment]
I have a similar problem. I am completely new to this and I am trying to run the "hello world" example given in "getting started" part of docs. I get this error
default_error
TypeError: Path must be a string. Received undefined
    at assertPath (path.js:7:11)
    at Object.join (path.js:468:7)
    at DefaultLocalizer.loadLocalePath (C:\Users\aravi\Desktop\cmdline\node_modules\botbuilder\lib\DefaultLocalizer.js:119:24)
    at C:\Users\aravi\Desktop\cmdline\node_modules\botbuilder\lib\DefaultLocalizer.js:104:27
    at C:\Users\aravi\Desktop\cmdline\node_modules\async\lib\async.js:181:20
    at iterate (C:\Users\aravi\Desktop\cmdline\node_modules\async\lib\async.js:262:13)
    at Object.async.forEachOfSeries.async.eachOfSeries (C:\Users\aravi\Desktop\cmdline\node_modules\async\lib\async.js:281:9)
    at Object.async.forEachSeries.async.eachSeries (C:\Users\aravi\Desktop\cmdline\node_modules\async\lib\async.js:214:22)
    at C:\Users\aravi\Desktop\cmdline\node_modules\botbuilder\lib\DefaultLocalizer.js:103:23
    at tryCallTwo (C:\Users\aravi\Desktop\cmdline\node_modules\promise\lib\core.js:45:5)

**Comment by KuhArt** [comment]
Maybe it is related to this commit.
https://github.com/Microsoft/BotBuilder/commit/85b02013c6816d1ffd3a37019146f651604b928b

**Comment by astrung** [comment]
same problem. How to fix it now ?

**Comment by KuhArt** [comment]
We change bot version to 3.1.0

28 окт. 2016 г. 11:34 пользователь "astrung" <notifications@github.com>
написал:

**Comment by astrung** [comment]
You can fix it now by add localization to your bot , until a new version fixes it.
Example: Replace

    var bot = new builder.UniversalBot(connector)

by 

    var bot = new builder.UniversalBot(connector, {
       localizerSettings: {
       botLocalePath: "./locale",
       defaultLocale: "en"
     }
     });

You can find about localization here : https://docs.botframework.com/en-us/node/builder/chat/localization/#navtitle

**Comment by KuhArt** [comment]
Wow. Thank you

28 окт. 2016 г. 11:39 пользователь "astrung" <notifications@github.com>
написал:
>
> You can fix it now by add localization to your bot , until a new version
fixes it.
> Example: Replace
>
> var bot = new builder.UniversalBot(connector)
>
> by
>
> var bot = new builder.UniversalBot(connector, { localizerSettings: {
botLocalePath: "./locale", defaultLocale: "en" } });
>
> You can find about localization here :
https://docs.botframework.com/en-us/node/builder/chat/localization/#navtitle

**Comment by jmendiara** [comment]
Related to #911

**Comment by bearnshaw** [comment]
Thanks @astrung, was having the same problem, fixed now!

**Comment by jdwilkin** [comment]
just to add... without setting the localizerSettings in the constructor as suggested above ALL of my localized strings were not being resolved.  I would see things like "list_or" and "default_response" in my chat output

**Comment by Stevenic** [comment]
Sorry guys... This is fixed in v3.4.1. There was a slight change in the generated output from the latest Typescript compiler update which caused this to surface.  My bad for not more thoroughly testing before releasing v3.4.

**Comment by ggoodale** [comment]
For those wondering why this doesn't work in the current release: `UniversalBot` doesn't take an object of type `IUniversalBotSettings` as the second constructor argument any more. Instead, do:
```
var bot = new builder.UniversalBot(connector);
bot.set('localizerSettings', {
    defaultLocale: 'en'
});
```
